### PR TITLE
Added LeakCanary to project

### PR DESCRIPTION
**Pull request by m-murad**
Fixes [TAVERNA-1021](https://issues.apache.org/jira/browse/TAVERNA-1021)

### CODE REVIEW

File: app/src/main/java/org/apache/taverna/mobile/TavernaApplication.java [hunk 1]
```diff
@@ -40,6 +41,12 @@ public void onCreate() {
         if (instance == null) {
             instance = this;
         }
+
+        if (LeakCanary.isInAnalyzerProcess(this)) {
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/TavernaApplication.java]: @m-murad 
Move it to bottom of the function.

Comment (m-murad) [inline app/src/main/java/org/apache/taverna/mobile/TavernaApplication.java]: @sagar15795 Sure

**Comment by m-murad** [comment]
@sagar15795 
Made the changes.

**Comment by stain** [comment]
Thanks for the pull request, @m-murad! 

So if I understand this right, the Taverna Mobile app released for users will just have a quiet `leakcanary:leakcanary-android-no-op` which doesn't do anything, while debug mode for developers with `leakcanary:leakcanary-android` will pop up if it can detect a memory leak?

BTW, have you been able to verify any such memory leaks?

**Comment by m-murad** [comment]
@stain

Yes that's how LeakCanary works.

I haven't been able to find a memory leak yet.

**Pull request merged**